src/plugins/mf_priority.cpp

```diff
@@ -331,6 +331,46 @@ static json_t *user_to_json (
 }
 
 
+/*
+ * Update the jobspec with the default bank the association used to
+ * submit their job under.
+ */
+static int update_jobspec_def_bank (flux_plugin_t *p, int userid)
+{
+    char *bank = NULL;
+    std::map<int, std::map<std::string, struct bank_info>>::iterator it;
+
+    it = users.find (userid);
+    if (it == users.end ()) {
+        return -1;
+    }
+
+    // look up default bank
+    bank = const_cast<char*> (users_def_bank[userid].c_str ());
+
+    json_t *bank_name = json_object ();
+    if (!bank_name)
+        return -1;
+
+    if (json_object_set_new (bank_name, "attributes.system.bank",
+                             json_string (bank)) < 0) {
+        json_decref (bank_name);
+        return -1;
+    }
+
+    // post jobspec-update event
+    if (flux_jobtap_event_post_pack (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "jobspec-update", "O",
+                                     bank_name) < 0) {
+        json_decref (bank_name);
+        return -1;
+    }
```

Comment:
I think you can use the new flux_jobtap_jobspec_update_pack() now instead.

https://github.com/flux-framework/flux-core/pull/5386